feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/TimetableItemDetailViewModel.kt

```diff
@@ -59,16 +63,20 @@ class TimetableItemDetailViewModel @Inject constructor(
                 isBookmarked = bookmarked,
             )
         }
+    val event: SharedFlow<TimetableItemDetailEvent> = _event
```

Comment:
Thanks! I think this is better because we don't have a leak.
But passing events will be difficult to manage. 
<img width="620" alt="image" src="https://github.com/DroidKaigi/conference-app-2023/assets/1386930/afb2a2d8-ff9b-4638-83c2-a3c5a3f94e57">
https://developer.android.com/topic/architecture/recommendations

How about "viewBookmarkRequestStateFlow = MutableStateFlow(RequestState.NotRequested)" and merge it to uiState 👀

Reply:
@takahirom 
> How about creating stateflow "viewBookmarkListRequestStateFlow = MutableStateFlow(RequestState.NotRequested)" and merge it to uiState 👀

That's sounds great!!
I have revised that policy.This is [the commit](https://github.com/DroidKaigi/conference-app-2023/pull/914/commits/ef3f57084c1811100347be396e8dee5807738a80).

Here is the intent of this commit.

> Why `val shouldViewBookmarkList` be defined at the top level of TimetableItemDetailScreenUiState instead of in Loaded?

Since I considered the screen transition request to be a separate axis concept not related to the current screen state, I thought it was inappropriate under Loaded.

> Why did I name it ViewBookmarkListRequestState instead of RequestState?

Suppose we name it RequestState. In this case, when the number of similar one-shot requests increases in the future
It will be implemented as one of the branches of RequestState.

Since UiState can only manage one RequestState at this time, we thought it would be difficult to send two one-shot requests at the same time.

If we named it "ViewBookmarkListRequestState", it would be used exclusively for screen transition requests to BookmarkScreen, so the above problem would be less likely to occur.


Translated with www.DeepL.com/Translator (free version)

Reply:
And it has also been verified to work properly!

<video src="https://github.com/DroidKaigi/conference-app-2023/assets/77588574/01a5977a-3beb-4a30-8c04-f01111e7034e" width="300" >